pkg/vsphere/compute/placement/random_test.go

```diff
@@ -62,47 +39,36 @@ func TestRandomRecommendHost(t *testing.T) {
 	hosts, err := cls.Hosts(op)
 	assert.NoError(t, err)
 
-	name := "test-vm"
-	vmx := fmt.Sprintf("%s/%s.vmx", name, name)
-	ds := sess.Datastore
-	secretKey, err := extraconfig.NewSecretKey()
-	if err != nil {
-		t.Fatal(err)
-	}
+	moref, err := test.CreateVM(op, sess, "test-vm")
+	assert.NoError(t, err)
 
-	spec := types.VirtualMachineConfigSpec{
-		Name:    name,
-		GuestId: string(types.VirtualMachineGuestOsIdentifierOtherGuest),
-		Files: &types.VirtualMachineFileInfo{
-			VmPathName: fmt.Sprintf("[%s] %s", ds.Name(), vmx),
-		},
-		ExtraConfig: []types.BaseOptionValue{
-			&types.OptionValue{
-				Key:   extraconfig.GuestInfoSecretKey,
-				Value: secretKey.String(),
-			},
-		},
-	}
+	v := vm.NewVirtualMachine(op, sess, moref)
 
-	res, err := tasks.WaitForResult(op, func(op context.Context) (tasks.Task, error) {
-		return sess.VMFolder.CreateVM(op, spec, sess.Pool, nil)
-	})
+	rhp, err := NewRandomHostPolicy(op, v.Session)
 	assert.NoError(t, err)
 
-	v := vm.NewVirtualMachine(op, sess, res.Result.(types.ManagedObjectReference))
-
-	rhp := NewRandomHostPolicy()
-	assert.False(t, rhp.CheckHost(op, v))
-	h, err := rhp.RecommendHost(op, v)
+	assert.False(t, rhp.CheckHost(op, nil))
+	h, err := rhp.RecommendHost(op, nil)
 	assert.NoError(t, err)
 
-	// TODO(jzt): come up with a better way to verify this than using a loop/moref comparison
+	top := h[0].Reference().String()
 	found := false
 	for _, host := range hosts {
-		if h.Reference().String() == host.Reference().String() {
+		if h[0].Reference().String() == host.Reference().String() {
 			found = true
+
+			// remove this host for the next test
+			h = append(h[:0], h[1:]...)
```

Comment:
Are you deliberately creating a new slice here, if so for what reason. I'm wondering why not:
```go
h = h[1:]
```

Reply:
Duh.. muscle memory. 👍